test/check-application

```diff
@@ -1184,6 +1185,86 @@ class TestApplication(testlib.MachineCase):
               .selectImageAndDownload() \
               .expectDownloadErrorForNonExistingTag()
 
+    def testPullImage(self):
+        b = self.browser
+        m = self.machine
+
+        self.execute(True, f"""
+            podman run -d -p 5000:5000 --name registry --stop-timeout 0 {IMG_REGISTRY}
+            podman run -d -p 6000:5000 --name registry_alt --stop-timeout 0 {IMG_REGISTRY}
+        """)
+
+        # Add local insecure registry into registries conf
+        m.write("/etc/containers/registries.conf", REGISTRIES_CONF)
+        self.execute(True, "systemctl stop podman.service")
+
+        self.execute(True, f"""
+            podman tag {IMG_BUSYBOX} localhost:5000/my-busybox
+            podman push localhost:5000/my-busybox
+            # Remove image with the same sha id
+            podman rmi localhost/test-busybox:latest
+        """)
+
+        new_container = 'new-container'
+        self.execute(True,
+                     f"podman run -d --name {new_container} --stop-timeout 0 localhost:5000/my-busybox touch /latest")
+        self.execute(True, f"podman commit {new_container} newimage")
+
+        new_image_sha = self.execute(True, "podman inspect --format '{{.Id}}' newimage").strip()
+        new_image_sel = f"#containers-images tbody tr[data-row-id=\"{new_image_sha}true\"]".lower()
+        busybox_id = self.execute(True, "podman inspect --format '{{.Id}}' localhost:5000/my-busybox").strip()
+
+        self.execute(True, f"""
+            podman push {new_image_sha} localhost:5000/my-busybox
+            podman push {new_image_sha} localhost:6000/my-busybox
+            podman rmi {new_image_sha}
+        """)
+
+        self.login()
+
+        b.wait_not_present(new_image_sel)
+        busybox_sel = f"#containers-images tbody tr[data-row-id=\"{busybox_id}true\"]".lower()
+
+        # Wait new image sha now visible
+        b.click("#containers-images button.pf-v5-c-expandable-section__toggle")
+        b.click(f'{busybox_sel} .pf-v5-c-menu-toggle')
+        b.click(busybox_sel + " button.pf-v5-c-menu__item:contains('Pull')")
+        b.wait_in_text("div.download-in-progress", "Pulling")
+        b.wait_not_present(busybox_sel)
+
+        # Updated image has a different sha
+        self.assertNotEqual(busybox_id,
+                self.execute(True, "podman inspect --format '{{.Id}}'  localhost:5000/my-busybox").strip())
+
+        # Try to pull multiple images, drop to superuser as the registry runs as superuser
+        b.drop_superuser()
```

Comment:
I don't understand this justification?

Reply:
The reason I run the tests as "normal" user is that you don't see superuser images which get in the way. (Basically the counting of `#container-images` becomes easier.)

Reply:
Ah, that makes it clear. The current comment totally doesn't explain that, can you please update it? Thanks!

Reply:
Done, to explain it better, only having images around which can be pulled makes the test not show any error dialogs as pulling `localhost/test-alpine` will return in an error.